Civi/Api4/Generic/BasicGetFieldsAction.php

```diff
@@ -220,22 +214,61 @@ public function getAction() {
     return $sub[$this->action] ?? $this->action;
   }
 
+  /**
+   * Resolve pseudoconstant options
+   *
+   * @param array $field
+   * @throws \CRM_Core_Exception
+   */
+  protected function getPseudoconstantOptions(array $field): array {
+    if (!empty($field['pseudoconstant']['optionGroupName'])) {
+      return $this->getOptionValues($field['pseudoconstant']['optionGroupName']);
+    }
+    if (!empty($field['pseudoconstant']['callback'])) {
+      return call_user_func(
+        \Civi\Core\Resolver::singleton()->get($field['pseudoconstant']['callback']),
+       $field['name'],
+       ['values' => $this->getValues()]
+      );
+    }
+    throw new \CRM_Core_Exception('Unsupported pseudoconstant type for field "' . $field['name'] . '"');
+  }
+
+  private function getOptionValues(string $optionGroupName): array {
+    $cacheKey = implode('_', [
+      \CRM_Core_Config::domainID(),
+      \CRM_Core_I18n::getLocale(),
+      'optionGroup',
+      $optionGroupName,
+    ]);
+    $optionValues = \Civi::cache('metadata')->get($cacheKey);
+
+    if (!is_array($optionValues)) {
+      $optionValues = \CRM_Core_OptionValue::getValues(['name' => $optionGroupName]);
+      \Civi::cache('metadata')->set($cacheKey, $optionValues);
+    }
+
+    // for field options, we want to use `value` key as the `id`
+    // (rather than global id from civicrm_option_value table)
+    foreach ($optionValues as &$option) {
+      $option['id'] = $option['value'];
+    }
+
+    return $optionValues;
+  }
+
   /**
    * Helper function to retrieve options from an option group (for non-DAO entities).
    *
    * @param string $optionGroupName
+   * @deprecated
    */
   public function pseudoconstantOptions(string $optionGroupName) {
```

Comment:
This seemed a poorly named function given it only deals with one type of pseudoconstant.

It's also strange as a public instance method on an Api action class. 

I didn't see any other callers in core and think it could probably be ripped out, but kept as deprecated for the time being.